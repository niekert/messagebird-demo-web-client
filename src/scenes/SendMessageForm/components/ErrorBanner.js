import React from 'react';
import { ERROR } from 'app-constants';
import { string } from 'prop-types';
import styled from 'styled-components';
import { prop } from 'styled-tools';
import Banner from 'shared/components/Banner';

const EmojiWrapper = styled.span`
  font-size: 25px;
  margin-right: ${prop('theme.spacing.0')};
`;

const ErrorBanner = ({ message }) => (
  <Banner type={ERROR}>
    <EmojiWrapper>
      <span role="img" aria-label="success">
        ðŸ˜”
      </span>
    </EmojiWrapper>
    {message}
  </Banner>
);

ErrorBanner.propTypes = {
  message: string.isRequired,
};

export default ErrorBanner;
