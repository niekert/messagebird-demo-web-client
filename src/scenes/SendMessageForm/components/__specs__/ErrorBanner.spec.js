import React from 'react';
import { shallow } from 'enzyme';
import ErrorBanner from '../ErrorBanner';

describe('<ErrorBanner />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(<ErrorBanner message="Krakaka" />);
    expect(wrapper).toMatchSnapshot();
  });
});
