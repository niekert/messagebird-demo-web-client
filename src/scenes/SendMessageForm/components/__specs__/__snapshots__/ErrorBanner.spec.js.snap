// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBanner /> renders correctly 1`] = `
<Banner
  className={null}
  durationMs={3000}
  type="ERROR"
>
  <ErrorBanner__EmojiWrapper>
    <span
      aria-label="success"
      role="img"
    >
      ðŸ˜”
    </span>
  </ErrorBanner__EmojiWrapper>
  Krakaka
</Banner>
`;
